src/panels/lovelace/cards/hui-logbook-card.ts

```diff
@@ -0,0 +1,313 @@
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFill: string[]
+  ) {
+    const includeDomains = ["light", "switch"];
+    const maxEntities = 3;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFill,
+      includeDomains
+    );
+
+    return {
+      entity: foundEntities,
+    };
+  }
+
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @internalProperty() private _config?: LogbookCardConfig;
+
+  @internalProperty() private _logbookEntries?: LogbookEntry[];
+
+  @internalProperty() private _persons = {};
+
+  @internalProperty() private _configEntities?: EntityConfig[];
+
+  private _lastLogbookDate?: Date;
+
+  private _throttleGetLogbookEntries = throttle(() => {
+    this._getLogBookData();
+  }, 10000);
+
+  public getCardSize(): number {
+    return 9 + (this._config?.title ? 1 : 0);
+  }
+
+  public setConfig(config: LogbookCardConfig): void {
+    this._configEntities = processConfigEntities<EntityConfig>(config.entities);
+
+    this._config = {
+      hours_to_show: 24,
+      ...config,
+    };
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    if (
+      changedProps.has("_config") ||
+      changedProps.has("_persons") ||
+      changedProps.has("_logbookEntries")
+    ) {
+      return true;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+
+    if (
+      !this._configEntities ||
+      !oldHass ||
+      oldHass.themes !== this.hass!.themes ||
+      oldHass.language !== this.hass!.language
+    ) {
+      return true;
+    }
+
+    for (const entity of this._configEntities) {
+      if (oldHass.states[entity.entity] !== this.hass!.states[entity.entity]) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  protected firstUpdated(): void {
+    this._fetchPersonNames();
+  }
+
+  protected updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const oldHass = changedProperties.get("hass") as HomeAssistant | undefined;
+    const oldConfig = changedProperties.get("_config") as LogbookCardConfig;
+
+    if (
+      !oldHass ||
+      !oldConfig ||
+      oldHass.themes !== this.hass.themes ||
+      oldConfig.theme !== this._config.theme
+    ) {
+      applyThemesOnElement(this, this.hass.themes, this._config.theme);
+    }
+
+    if (
+      changedProperties.has("_config") &&
+      (oldConfig?.entities !== this._config.entities ||
+        oldConfig?.hours_to_show !== this._config!.hours_to_show)
+    ) {
+      this._logbookEntries = undefined;
+      this._lastLogbookDate = undefined;
+
+      if (!this._configEntities) {
+        return;
+      }
+
+      this._throttleGetLogbookEntries();
+      return;
+    }
+
+    if (
+      oldHass &&
+      this._configEntities!.some(
+        (entity) =>
+          "entity" in entity &&
```

Comment:
Why have this here but not in shouldupdate?

Reply:
And do we need this check here if we already did it in `shouldUpdate`?

Reply:
I think we need both. Since to get to updated it can be more than just if the state obj changed